line_detection/src/backproject.py

```diff
@@ -0,0 +1,369 @@
+#!/usr/bin/env python
+import sys
+import numpy as np
+import cv2
+import math
+#import time
+import rospy
+import sensor_msgs
+from sensor_msgs.msg import CompressedImage
+from sensor_msgs.msg import Image
+import rospkg
+from dynamic_reconfigure.server import Server
+from line_detection.cfg import LineDetectionConfig
+from cv_bridge import CvBridge, CvBridgeError
+
+###############################################################################
+## Chicago Engineering Design Team
+## Line Detection Example using Python OpenCV for autonomous robot Scipio
+##    (IGVC competition).
+## @author Basheer Subei
+## @email basheersubei@gmail.com
+#######################################################
+##
+## Gabor Filter example
+##
+###############################################################################
+
+
+class line_detection:
```

Comment:
(Pep8)[https://www.python.org/dev/peps/pep-0008/#class-names] would encourage LineDetection as the class name and I would encourage you to use the OO syntax "LineDetection(object):"